### PR TITLE
refactor utils

### PR DESCRIPTION
**Pull request by mateeullahmalik**

### CODE REVIEW

File: utils/utils.go [hunk 1]
```diff
@@ -26,23 +26,27 @@ import (
...
+		if err := os.MkdirAll(path, 0755); err != nil {
 			log.WithContext(ctx).WithError(err).Error("Error creating directory")
 			return errors.Errorf("Failed to create directory: %v", err)
 		}
 		log.WithContext(ctx).Infof("Directory created on %s", path)
-	} else {
+
+		return nil
+	}
+	switch force {
```
Comment (georgkcode0) [inline utils/utils.go:38]: better if we should use `if`

File: utils/utils.go [hunk 2]
```diff
@@ -26,23 +26,27 @@ import (
...
-			if err != nil {
-				log.WithContext(ctx).WithError(err).Error("Error creating directory")
-				return errors.Errorf("Failed to create directory: %v", err)
+			if err := create(path); err != nil {
+				return err
 			}
-			log.WithContext(ctx).Infof("Directory created on %s \n", path)
 		} else {
-			return errors.Errorf("Directory already exists on %s", path)
+			return errors.Errorf("directory already exists on %s", path)
```
Comment (georgkcode0) [inline utils/utils.go:49]: consider to `return fs.ErrExist`

File: utils/utils.go [hunk 3]
```diff
@@ -52,33 +56,34 @@ func CreateFolder(ctx context.Context, path string, force bool) error {
...
-		if os.IsNotExist(err) {
-			var file, err = os.Create(fileName)
-			if err != nil {
-				log.WithContext(ctx).WithError(err).Error("Error creating file")
-				return "", errors.Errorf("Failed to create file: %v", err)
+		log.WithContext(ctx).Infof("File created: %s \n", fileName)
+		return nil
+	}
+
+	switch force {
```
Comment (georgkcode0) [inline utils/utils.go:71]: should use `if`

File: utils/utils.go [hunk 4]
```diff
@@ -277,18 +277,14 @@ func Untar(dst string, r io.Reader, filenames ...string) error {
 
 // Unzip will decompress a zip archive, moving all files and folders
 // within the zip file (parameter 1) to an output directory (parameter 2).
-func Unzip(src string, dest string, fPaths ...string) ([]string, error) {
-
-	var filenames []string
-
+func Unzip(src string, dest string, fPaths ...string) (filenames []string, err error) {
```
Comment (georgkcode0) [inline utils/utils.go:280]: Please take a look at [PR14](https://github.com/pastelnetwork/pastel-utility/pull/14). it's seem @tsimafeisetchinava000 remove this part

```
if !Contains(fPaths, fpath) && len(fPaths) != 0 {
	continue
}
```

File: utils/utils.go (hunk 2 above)
Comment (georgkcode0) [reply utils/utils.go:49]: or define a new const error

File: utils/utils.go (hunk 4 above)
Comment (mateeullahmalik) [reply utils/utils.go:280]: PR # 14 is still open. The changes will apply in the scope of that PR

File: utils/utils.go (hunk 1 above)
Comment (mateeullahmalik) [reply utils/utils.go]: changes made, Thanks!

**Review by georgkcode0 — APPROVED** [review]
LGTM

**Pull request merged**